### PR TITLE
05875 query workflow improvements

### PR DESCRIPTION
**Pull request by netopyr**
Simplifications of the query workflow that I already made available. Had to redo them, because squash&merge does not work well with switching base branches.

- Removed `QueryContext`
- Removed metrics in `QueryWorkflowImpl`
- Introduced `NodeInfo`, `NetworkInfo`, and `CurrentPlatformStatus` in hedera-app and provided mono adapters
- Some minor simplifications in `QueryWorkflowImpl`

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package com.hedera.node.app.spi.info;
+
+import com.swirlds.common.system.PlatformStatus;
+
+/**
+ * Provides the current platform status.
+ */
+public interface CurrentPlatformStatus {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:25]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java [hunk 2]
```diff
@@ -5,22 +5,30 @@
...
 
 import com.hedera.pbj.runtime.io.buffer.Bytes;
 
 /**
- * Provides context for query processing.
+ * Provides information about the network.
  */
-public interface QueryContext {
-    Bytes getLedgerId();
+public interface NetworkInfo {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:25]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java [hunk 3]
```diff
@@ -5,22 +5,30 @@
...
+
+    /**
+     * Returns the current ledger ID.
+     *
+     * @return the {@link Bytes} of the current ledger ID
+     */
+    Bytes ledgerId();
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:32]: NotNull annotation

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+package com.hedera.node.app.spi.info;
+
+import com.swirlds.common.system.address.AddressBook;
+
+/**
+ * Summarizes useful information about the nodes in the {@link AddressBook} from the Platform. In
+ * the future, there may be events that require re-reading the book; but at present nodes may treat
+ * the initializing book as static.
+ */
+public interface NodeInfo {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:27]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Convenience method to check if this node is zero-stake.
+     *
+     * @return whether this node has zero stake.
+     */
+    boolean isSelfZeroStake();
```
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:34]: NotNull annotation

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+public class MonoCurrentPlatformStatus implements CurrentPlatformStatus {
+
+    private final com.hedera.node.app.service.mono.context.CurrentPlatformStatus delegate;
+
+    /**
+     * Constructs a {@link MonoCurrentPlatformStatus} with the given delegate.
+     *
+     * @param delegate the delegate
+     */
+    public MonoCurrentPlatformStatus(com.hedera.node.app.service.mono.context.CurrentPlatformStatus delegate) {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java:35]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.hedera.node.app.spi.info.NetworkInfo;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+
+public class MonoNetworkInfo implements NetworkInfo {
+
+
+    private final com.hedera.node.app.service.mono.config.NetworkInfo delegate;
+
+    public MonoNetworkInfo(com.hedera.node.app.service.mono.config.NetworkInfo delegate) {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java:28]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+public class MonoNodeInfo implements NodeInfo {
+
+    private final com.hedera.node.app.service.mono.context.NodeInfo delegate;
+
+    /**
+     * Constructs a {@link MonoNodeInfo} with the given delegate.
+     *
+     * @param delegate the delegate
+     */
+    public MonoNodeInfo(com.hedera.node.app.service.mono.context.NodeInfo delegate) {
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:34]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java [hunk 9]
```diff
@@ -277,6 +219,15 @@ public void handleQuery(
         }
     }
 
+    private Query parseQuery(Bytes requestBuffer) {
+        try {
+            return queryParser.parseStrict(requestBuffer.toReadableSequentialData());
+        } catch (IOException e) {
+            // TODO there may be other types of errors here. Please cross check with ingest parsing
+            throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:227]:
```suggestion
            throw new StatusRuntimeException(Status.INVALID_ARGUMENT, e);
```

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java [hunk 10]
```diff
@@ -165,7 +169,7 @@ private Optional<ConsensusTopicInfo> infoForTopic(
                     .ifPresent(account ->
                             info.autoRenewAccount(AccountID.newBuilder().accountNum(account)));
 
-            info.ledgerId(queryContext.getLedgerId());
+            info.ledgerId(networkInfo.ledgerId());
```
Comment (hendrikebbers) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:172]: networkInfo can be null

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Returns the current platform status.
+     *
+     * @return the current platform status
+     */
+    PlatformStatus get();
```
[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:32]: Fixed

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:25]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java (hunk 2 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:25]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java (hunk 3 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:32]: Fixed

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java (hunk 4 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:27]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java (hunk 5 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:34]: I see a pattern. Is Chat GPT making your code reviews? 🙂

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java (hunk 6 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java:35]: Fixed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java (hunk 7 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java:28]: Fixed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java (hunk 9 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:227]: `StatusRuntimeException` does not allow this.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java (hunk 10 above)
Comment (netopyr) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:172]: It should not. A `networkInfo` always has to be provided during construction. The empty constructor is just left in there for now to make the project compile.

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Pull request merged**